iceoryx2/src/service/mod.rs

```diff
@@ -369,11 +369,19 @@ pub struct ServiceDetails<S: Service> {
 /// Represents the [`Service`]s state.
 #[derive(Debug)]
 pub struct ServiceState<S: Service, R: ServiceResource> {
+    // For this struct it is important to know that Rust drops fields of a struct in declaration
+    // order - not in memory order!
```

Comment:
Maybe it's just because it's late but if we would not have talked about this earlier, I would not have understood the comment. How about this?
```suggestion
    // order - not in reverse order!
```